src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -970,10 +989,18 @@ private bool ImportIntrinsicCall(MethodDesc method)
             return false;
         }
 
-        private void HandleCall(MethodDesc callee, ILOpcode opcode = ILOpcode.call)
-        { 
-            AddMethodReference(callee);
-            int offset = GetTotalParameterOffset() + GetTotalLocalOffset() + callee.Signature.ReturnType.GetElementSize().AsInt;
+        private void HandleCall(MethodDesc callee, MethodSignature signature, ILOpcode opcode = ILOpcode.call, LLVMValueRef calliTarget = default(LLVMValueRef))
+        {
+            if (opcode == ILOpcode.callvirt && callee.IsVirtual)
+            {
+                AddVirtualMethodReference(callee);
```

Comment:
Btw: I'm not sure the `AddVirtualMethodReference` calls are needed in the importer at all. These dependencies should fall out from having a reloc to `WebAssemblyVTableSlotNode` (which in turn depends on the `VirtualMethodUse` if we're tracking virtual method use).

I think this is just another wart taken from CppCodgen (a codegen design that predates the dependency node driven architecture of the compiler).

Reply:
Removing this seems to have been ok.